lib/screens/study_nurse_screen/configuration_screen/configuration_pages.dart

```diff
@@ -0,0 +1,135 @@
+}
+
+class _ConfigurationPages extends State<ConfigurationPages> {
+  PageController controller = PageController();
+
+  final List<Widget> _list = <Widget>[
+    const ConfigurationForm(),
+    const ConfigurationVitalValues(),
+    const ConfigurationActivityAndRest(),
+    const ConfigurationBodyAndMind(),
+    const ConfigurationMedicationAndTherapy(),
+    const ConfigurationSummary()
+  ];
+
+  int _currentPage = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    final GlobalTheme theme = Theme.of(context).extension<GlobalTheme>()!;
+
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: AppBar(
+        title: Text(AppLocalizations.of(context)!.configuration),
+        backgroundColor: theme.darkGreen3,
+      ),
+      body: Form(
+        key: formKeyConfiguration,
+        child: PageView(
+          scrollDirection: Axis.vertical,
+          controller: controller,
+          onPageChanged: (int num) {
+            setState(() {
+              _currentPage = num;
+            });
+          },
+          children: _list,
+        ),
+      ),
+      bottomNavigationBar: Row(
+        children: <Widget>[
+          Expanded(
+            child: PicosInkWellButton(
+              text: AppLocalizations.of(context)!.back,
+              onTap: () {
+                if (_currentPage == 0) {
+                  Navigator.of(context).pop();
+                } else {
+                  controller.jumpToPage(_currentPage - 1);
+                }
+              },
+              buttonColor1: theme.grey3,
+              buttonColor2: theme.grey1,
+            ),
+          ),
+          Expanded(
+            child: PicosInkWellButton(
+              text: AppLocalizations.of(context)!.proceed,
+              onTap: () async {
+                if (_currentPage == _list.length - 1) {
+                  formKeyConfiguration.currentState!.save();
+                  bool backendCreatePatient = await Backend.createPatient();
+                  if (backendCreatePatient) {
+                    if (!mounted) return;
+                    Navigator.of(context)
+                        .pushNamed('/configuration-finish-screen');
+                  } else {
+                    log('Fehler beim Anlegen des Patienten!');
+                  }
+                } else {
+                  if (_currentPage == 0) {
+                    if (formKeyConfiguration.currentState!.validate()) {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        SnackBar(
+                          content: Text(
+                            AppLocalizations.of(context)!.submitData,
+                          ),
+                        ),
+                      );
+                      controller.jumpToPage(_currentPage + 1);
+                    }
+                  } else {
+                    controller.jumpToPage(_currentPage + 1);
+                  }
+                }
+              },
```

Comment:
```suggestion
                if (_currentPage != _list.length - 1) {
                  if (_currentPage == 0 &&
                      formKeyConfiguration.currentState!.validate()) {
                    ScaffoldMessenger.of(context).showSnackBar(
                      SnackBar(
                        content: Text(
                          AppLocalizations.of(context)!.submitData,
                        ),
                      ),
                    );
                  }

                  controller.jumpToPage(_currentPage + 1);
                  return;
                }

                formKeyConfiguration.currentState!.save();
                bool backendCreatePatient = await Backend.createPatient();

                if (!backendCreatePatient) {
                  log('Fehler beim Anlegen des Patienten!');
                  return;
                }

                if (!mounted) {
                  return;
                }

                Navigator.of(context).pushNamed('/configuration-finish-screen');
              },
```

Suggestion how it could be clearer.

Reply:
Your suggestion doesn't work unfortunately.

Reply:
Yeah, I've noticed a mistake I made. But that should proof my point that this part of the code is hard to follow and has to be made clearer.
Btw. is there a particular reason why you are using a SnackBar displaying a message suggesting that the data has been transmitted? I mean, at this point nothing has been transmitted and it would give the user a false sense, telling him that.

Reply:
Yeah thanks, it works now.